backend/experiment/actions/final.py

```diff
@@ -1,45 +1,174 @@
+        total_score (float): Explicit final score. If None, this is derived from the session.
+        logo (LogoDto): Optional logo configuration for branding. For example:
+                        {"image": "/static/logo.png", "link": "https://example.com"}.
 
-    Relates to client component: Final.js
+    Attributes:
+        session (Session): The session object for the current participant.
+        title (str): The title for the final view.
+        final_text (str): A concluding message for the participant.
+        button (ButtonDto): Configuration for an optional CTA button.
+        rank (str): The participant's performance rank.
+        show_profile_link (bool): Whether to show a profile link.
+        show_participant_link (bool): Whether to show participant-related data.
+        show_participant_id_only (bool): Whether only to display the participant ID.
+        feedback_info (dict): Information for a feedback section.
+        logo (LogoDto): Configuration for a logo displayed in the final view.
+        total_score (float): The participant’s final score.
+        points (str): Label for the scoring unit.
+
+    Note:
+        The `action()` method returns a `FinalActionDto` that can be consumed by the frontend
+        to render the final screen.
     """
 
-    ID = 'FINAL'
+    ID = "FINAL"
 
     RANKS = {
-        'PLASTIC': {'text': _('plastic'), 'class': 'plastic'},
-        'BRONZE':  {'text': _('bronze'), 'class': 'bronze'},
-        'SILVER': {'text': _('silver'), 'class': 'silver'},
-        'GOLD': {'text': _('gold'), 'class': 'gold'},
-        'PLATINUM': {'text': _('platinum'), 'class': 'platinum'},
-        'DIAMOND': {'text': _('diamond'), 'class': 'diamond'}
+        "PLASTIC": {"text": _("plastic"), "class": "plastic"},
+        "BRONZE": {"text": _("bronze"), "class": "bronze"},
+        "SILVER": {"text": _("silver"), "class": "silver"},
+        "GOLD": {"text": _("gold"), "class": "gold"},
+        "PLATINUM": {"text": _("platinum"), "class": "platinum"},
+        "DIAMOND": {"text": _("diamond"), "class": "diamond"},
     }
 
     def __init__(
         self,
         session: Session,
         title: str = _("Final score"),
-        final_text: str = None,
-        button: dict = None,
-        points: str = None,
-        rank: str = None,
+        final_text: str | None = None,
+        button: ButtonDto | None = None,
+        points: str | None = None,
+        rank: str | None = None,
         show_profile_link: bool = False,
         show_participant_link: bool = False,
         show_participant_id_only: bool = False,
-        feedback_info: dict = None,
-        total_score: float = None,
-        logo: dict = None,
+        feedback_info: dict | None = None,
+        total_score: float | None = None,
+        logo: LogoDto | None = None,
     ):
+        """
+        Initialize the final action data.
 
+        Args:
```

Comment:
No, don't repeat this again in the `__init__` method, I would say. It's sufficient to add the arguments to the class, perhaps add an example implementation there. It would clutter the documentation page if we specify the arguments in two (or even three!) places.